freqtrade/freqai/data_kitchen.py

```diff
@@ -0,0 +1,1205 @@
+        else:
+            dataframe = base_dataframes[self.config["timeframe"]].copy()
+
+        sgi = False
+        for tf in tfs:
+            if tf == tfs[-1]:
+                sgi = True  # doing this last allows user to use all tf raw prices in labels
+            dataframe = strategy.populate_any_indicators(
+                pair,
+                dataframe.copy(),
+                tf,
+                informative=base_dataframes[tf],
+                set_generalized_indicators=sgi
+            )
+            if pairs:
+                for i in pairs:
+                    if pair in i:
+                        continue  # dont repeat anything from whitelist
+                    dataframe = strategy.populate_any_indicators(
+                        i,
+                        dataframe.copy(),
+                        tf,
+                        informative=corr_dataframes[i][tf]
+                    )
+
+        return dataframe
+
+    def fit_labels(self) -> None:
+        """
+        Fit the labels with a gaussian distribution
+        """
+        import scipy as spy
+
+        self.data["labels_mean"], self.data["labels_std"] = {}, {}
+        for label in self.data_dictionary["train_labels"].columns:
+            if self.data_dictionary["train_labels"][label].dtype == object:
+                continue
+            f = spy.stats.norm.fit(self.data_dictionary["train_labels"][label])
+            self.data["labels_mean"][label], self.data["labels_std"][label] = f[0], f[1]
+
+        # KEEPME incase we want to let user start to grab quantiles.
+        # upper_q = spy.stats.norm.ppf(self.freqai_config['feature_parameters'][
+        #                                                   'target_quantile'], *f)
+        # lower_q = spy.stats.norm.ppf(1 - self.freqai_config['feature_parameters'][
+        #                                                       'target_quantile'], *f)
+        # self.data["upper_quantile"] = upper_q
+        # self.data["lower_quantile"] = lower_q
+        return
+
+    def remove_features_from_df(self, dataframe: DataFrame) -> DataFrame:
+        """
+        Remove the features from the dataframe before returning it to strategy. This keeps it
+        compact for Frequi purposes.
+        """
+        to_keep = [
+            col for col in dataframe.columns if not col.startswith("%") or col.startswith("%%")
+        ]
+        return dataframe[to_keep]
+
+    def get_current_trade_database(self) -> None:
+
+        if self.database_path is None:
+            logger.warning('No trade database found. Skipping analysis.')
+            return
+
+        data = sqlite3.connect(self.database_name)
+        query = data.execute("SELECT * From trades")
+        cols = [column[0] for column in query.description]
+        df = pd.DataFrame.from_records(data=query.fetchall(), columns=cols)
+        self.trade_database_df = df.dropna(subset='close_date')
+        data.close()
+
+    def np_encoder(self, object):
```

Comment:
I don't think this method is used (not from datakitchen, anyway).

so it can either be removed - or refactored into a "non-class" method - so it can be used wherever needed.